ida-ws/src/main/java/upb/ida/logging/AspectLogger.java

```diff
@@ -0,0 +1,143 @@
+
+	@Autowired(required=true)
+	private HttpServletRequest request;
+	@Autowired
+	private ResponseBean response;
+	private Logger requestReponseLogger;
+	private Logger exceptionLogger;
+	private Logger datasetResponseLogger;
+	
+    /**
+     * Constructor of AspectLogger class
+     */
+	AspectLogger()
+	{
+		BasicConfigurator.configure();
+		String log4jConfigFile = System.getProperty("user.dir") + File.separator + "log4j.properties";
+		PropertyConfigurator.configure(log4jConfigFile);
+		requestReponseLogger = Logger.getLogger("RequestResponseLogger");
+		exceptionLogger = Logger.getLogger("ExceptionLogger");
+		datasetResponseLogger = Logger.getLogger("DatasetResponseLogger");
+	}
+	
+	@Pointcut("execution(* upb.ida.rest.MessageRestController*.*(..))")
+	public void controller() {
+	}
+
+    /**
+     * Method to log the Request to Message Rest Controller
+     * @param joinPoint
+     */
+    @Before("controller()")
+    public void logBeforeMethod(JoinPoint joinPoint) {
+        StringBuffer logMessage = new StringBuffer();
+        logMessage.append("Request - ");
+        logMessage.append(request.getRemoteAddr());
+        logMessage.append(" ");
+        logMessage.append(joinPoint.getTarget().getClass().getName());
+        logMessage.append(".");
+        logMessage.append(joinPoint.getSignature().getName());
+        logMessage.append("(");
+        Object[] args = joinPoint.getArgs();
+        for (int i = 0; i < args.length; i++) {
+            logMessage.append(args[i]).append(",");
+        }
+        if (args.length > 0) {
+            logMessage.deleteCharAt(logMessage.length() - 1);
+        }
+
+        logMessage.append(")");
+        requestReponseLogger.info(logMessage.toString());
+    }
+    
+
+    /** Method to log the response by Message Rest Controller
+    * @param joinPoint
+    * @param retVal
+    */
+   @AfterReturning(pointcut = "controller()", returning = "retVal")
+   public void logAfterReturningMethod(JoinPoint  joinPoint, Object retVal) {
+       StringBuffer logMessage = new StringBuffer();
+       logMessage.append("Response - ");
+       logMessage.append(request.getRemoteAddr());
+       logMessage.append(" ");
+       logMessage.append(joinPoint.getTarget().getClass().getName());
+       logMessage.append(".");
+       logMessage.append(joinPoint.getSignature().getName());
+       logMessage.append("(");
+       // append args
+       Object[] args = joinPoint.getArgs();
+       for (int i = 0; i < args.length; i++) {
+           logMessage.append(args[i]).append(",");
+       }
+       if (args.length > 0) {
+           logMessage.deleteCharAt(logMessage.length() - 1);
+       }
+       logMessage.append(")");
+       logMessage.append(" return Value :");
+       ResponseBean returnBean = (ResponseBean) retVal;
+       if(returnBean.getActnCode() == IDALiteral.UIA_DTTABLE)
+       {
+    	   logMessage.append("Dataset");
+           requestReponseLogger.info(logMessage.toString());
```

Comment:
this line is in both if and else, maybe you can put it out

Reply:
the logic is different in there.

Reply:
explain how it's different if I just put the `requestReponseLogger.info(logMessage.toString());` out the if and else block and at the end?

Reply:
Actually in the if-statement, it is logging the log message + "Dataset" to the requestresponse log file and the response bean to the dataset log file. While in the else-statement for the normal flow, it is logging the log message + return bean in the requestresponse log file.